### PR TITLE
[BUGFIX release] Publish builds to s3 on release & beta channels even

### LINKED ISSUE #17297: release.json not updating
https://s3.amazonaws.com/builds.emberjs.com/release.json references 3.4.4 and is not updating. This is used by ember-try and `getChannelURL`

### PR DESCRIPTION
**Pull request by kategengler**
when there is a tag for the commit.

- Always publish as a tag when publishing locally (vs on Travis)

Fixes #17297

### CODE REVIEW

File: config/s3ProjectConfig.js [hunk 1]
```diff
@@ -63,12 +63,6 @@ function fileObject(baseName, extension, contentType, currentRevision, tag, date
     },
   };
 
-  if (tag) {
```
Comment (rwjblue) [inline config/s3ProjectConfig.js]: 👍 - This was never used (because we only publish to S3 for non-tagged builds).

File: broccoli/build-info.js [hunk 2]
```diff
@@ -34,9 +34,9 @@ function buildInfo(options) {
 function buildGitInfo(root) {
   let info = gitRepoInfo(root);
   return {
-    sha: process.env.TRAVIS_COMMIT || info.sha,
-    branch: process.env.TRAVIS_BRANCH || info.branch,
-    tag: process.env.TRAVIS_TAG || info.tag,
+    sha: info.sha,
+    branch: process.env.TRAVIS_BRANCH || info.branch, // Travis builds are always detached
```
Comment (rwjblue) [inline broccoli/build-info.js]: What is `info.branch` when detached? 

I kinda feel like we may want to put all the `TRAVIS` smarts in `buildFromParts` directly. What do you think?

**Review by rwjblue — COMMENTED** [review]
Overall, it looks good to me (and should solve the issues we have been dealing with), only one inline question/comment...

File: broccoli/build-info.js (hunk 2 above)
Comment (kategengler) [reply broccoli/build-info.js]: `info.branch` when detached is null.

Comment (kategengler) [reply broccoli/build-info.js]: @rwjblue Can you restate or clarify "I kinda feel like we may want to put all the TRAVIS smarts in buildFromParts directly. What do you think?"?

Comment (rwjblue) [reply broccoli/build-info.js]: Basically, just that this could be `branch: info.branch` and down below where all the other travis detection logic.

Comment (kategengler) [reply broccoli/build-info.js]: 👍

**Review by rwjblue — APPROVED** [review]

**Pull request merged**